### PR TITLE
fix: use ajv for schema ref resolving

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)
      
This PR shows that all Fastify tests work with https://github.com/fastify/fast-json-stringify-compiler/pull/14 and https://github.com/fastify/fast-json-stringify/pull/462 changes.

### CODE REVIEW

**Comment by ivan-tymoshenko** [comment]
@climba03003 Can you help with CI. I guess it is related to https://github.com/fastify/fastify/pull/3997. Some Ci works, some of them fail. Everything is ok on my local.

**Comment by climba03003** [comment]
It is strange that it pass `node@16` and `node@18` but not `node@14`

**Comment by climba03003** [comment]
No idea why it's fail. 
I tested against `node@14` with `Powershell`. No error exist.
```
PS fastify> node -v
v14.19.3
PS fastify> node .\test\build\error-serializer.test.js
TAP version 13
# Subtest: check generated code syntax
    1..1
    ok 1 - should be equal
ok 1 - check generated code syntax # time=992.822ms

# Subtest: ensure the current error serializer is latest
    1..1
    ok 1 - should be equal
ok 2 - ensure the current error serializer is latest # time=1.4ms

1..2
# time=998.354ms
```

File: package.json [hunk 1]
```diff
@@ -146,7 +146,7 @@
     "eslint-plugin-n": "^15.2.0",
     "eslint-plugin-promise": "^6.0.0",
     "fast-json-body": "^1.1.0",
-    "fast-json-stringify": "^4.2.0",
+    "fast-json-stringify": "ivan-tymoshenko/fast-json-stringify#use-ajv-for-schema-ref-resolving",
```
Comment (mcollina) [inline package.json:149]:
```suggestion
    "fast-json-stringify": "^5.0.0",
```

File: package.json [hunk 2]
```diff
@@ -179,7 +179,7 @@
   "dependencies": {
     "@fastify/ajv-compiler": "^3.1.0",
     "@fastify/error": "^3.0.0",
-    "@fastify/fast-json-stringify-compiler": "^3.0.1",
+    "@fastify/fast-json-stringify-compiler": "ivan-tymoshenko/fast-json-stringify-compiler#use-ajv-for-schema-ref-resolving",
```
Comment (mcollina) [inline package.json:182]:
```suggestion
    "@fastify/fast-json-stringify-compiler": "^4.0.0",
```

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**